src/componententity.cpp

```diff
@@ -279,8 +276,6 @@ void ComponentEntity::removeUnits(size_t index)
 {
     if (index < mPimpl->mUnits.size()) {
```

Comment:
What if `index` is lower than zero? Also, I thought (from #140) that we had agreed to return a status (`true`, `false` or `nullptr`)? I would therefore expect to have something like:

```
bool ComponentEntity::removeUnits(size_t index)
{
     if ((index >= 0) && (index < mPimpl->mUnits.size())) {
        mPimpl->mUnits.erase(mPimpl->mUnits.begin() + index);

        return true;
    } else {
        return false;
    }
}
```

The same holds for all the functions where we don't throw an exception anymore.